source/dub/recipe/packagerecipe.d

```diff
@@ -74,20 +76,33 @@ string getSubPackageName(string package_name) @safe pure
 */
 struct PackageRecipe {
 	string name;
+	@Optional @Name("version")
 	string version_;
+	@Optional
 	string description;
+	@Optional
 	string homepage;
+	@Optional
 	string[] authors;
+	@Optional
 	string copyright;
+	@Optional
 	string license;
+	@Optional @Name("-ddoxFilterArgs")
 	string[] ddoxFilterArgs;
+	@Optional @Name("-ddoxTool")
 	string ddoxTool;
-	BuildSettingsTemplate buildSettings;
+
+	@Optional BuildSettingsTemplate buildSettings;
+	alias buildSettings this;
```

Comment:
unrelated change, but I think this is useful to have

Reply:
It is related. `BuildSettingsTemplate` are flattened everywhere they are used. The way to let configy know a field is flattened is by using `alias this`. Another approach would have been to use a `mixin template` for the fields but that was too invasive.
Without the `alias this`, configy would allow a field named `buildSettings`, but not allow a field name `targetType` at the top level.